### PR TITLE
[Feature][Kimi K3] Support MRV2 GQA DSpark reduced-expert validation

### PR DESCRIPTION
**Pull request by qijiajin**
## What this PR does

- adds runtime-only Kimi K3 routed-expert reduction (default remains the full checkpoint)
- establishes the materialized auxiliary-hidden-state contract required by legacy Qwen3 GQA DSpark
- keeps the full 93 target layers while allowing only routed expert weights to be reduced
- normalizes grouped Mamba cache specs for MRV2 and keeps Ascend MLA/indexer caches in the block-zeroing lifecycle
- aligns the MLA cache-shape API with current vLLM
- fixes QuaRot GQA draft loading so the BF16 draft receives target rotation metadata without inheriting the target quantization config
- aligns the draft FC, embedding, and lm_head before upstream decides whether to share target weights
- adds a fail-fast ownership postcondition for QuaRot GQA drafts and diagnostic logging for legacy query/RoPE layout semantics

## Root cause fixed

The GQA draft checkpoint does not contain its own embedding or lm_head. Upstream correctly clears the target quantization config for the BF16 draft, but the previous Ascend loader also used that cleared config to decide whether QuaRot metadata existed. It therefore skipped the full QuaRot load path, and upstream aliased the rotated target embedding/lm_head into the unquantized draft. The MRV2 override only rotated FC after sharing had already happened.

The new path passes only the target rotation metadata into draft construction, reuses the existing Qwen3 DSpark loader to align all three boundaries, removes the post-load FC-only workaround, and rejects accidental target weight sharing.

## Validation

- `48 passed` across focused Kimi K3, DSpark config/model/speculator, and MRV2 attention/KV-cache tests
- `py_compile`
- `git diff --check`
- single-node TP16/EP16, MRV2 eager, full 93 layers, runtime 16/896 routed experts, K=7
- all 16 ranks reported `embed_shared=False`, `lm_head_shared=False`, `has_own_embed=True`, and `has_own_lm_head=True`
- deterministic short request: acceptance improved from 0% to 26.2%
- one real GSM8K row, request-only Prometheus counter delta: 111 accepted / 161 drafted = 68.94%; all seven draft positions had non-zero acceptance
- service startup, `/health`, `/v1/models`, and real completion requests returned successfully

### Exact 16-expert Target/DSpark token parity

A separate C/D oracle test used fresh service starts so both variants began with an empty prefix cache:

- C: 16-expert Target-only
- D: the same 16-expert Target plus GQA DSpark, K=7
- same Target tokenizer and served model, fixed 63 integer `prompt_token_ids`
- identical payload: `add_special_tokens=false`, `temperature=0`, `top_p=1`, `seed=0`, `max_tokens=128`, empty `stop` and `stop_token_ids`, `ignore_eos=false`
- `/v1/completions` with `return_token_ids=true`; no chat template or parser
- returned prompt IDs: 63/63 exactly equal
- final generated token IDs: **128/128 exactly equal**
- `first_mismatch_index=null`, identical `finish_reason=length`, and identical decoded text

This passes the core speculative-correctness check for the fixed input: after the QuaRot/layout fix, enabling GQA DSpark did not change the current reduced-expert Target's final token sequence.

## Validation boundary

The reduced-expert mode does not edit `config.json` or checkpoint weights. The checkpoint remains 93 layers and 896 experts; only runtime expert instantiation/loading is capped at 16. This mode is intended for functional pipeline validation, not model-quality evaluation.

The GSM8K response did not match the reference answer and repeated text in both Target-only and DSpark runs. Exact C/D parity shows that this output is produced by the current 16-expert Target itself rather than being introduced by speculative decoding. It does not prove Target accuracy. A full-expert Target-only baseline and then full-expert DSpark comparison are still required before making an accuracy claim.

The parity result covers one fixed 63-token prompt and 128 generated tokens. It is not a full-expert, multi-sample, complete GSM8K, or graph-mode accuracy result.

The legacy config omits explicit `sample_from_anchor` semantics and currently defaults to `true`. K remains 7; this PR does not change it to 6.

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/worker/v2/spec_decode/dspark/speculator.py [hunk 1]
```diff
@@ -65,6 +89,63 @@ def load_draft_model(
...
+    def _configure_target_aux_hidden_state_format(
+        target_model: torch.nn.Module,
+        format_provider: Any,
+    ) -> str | None:
+        """Apply a draft-declared auxiliary-hidden-state representation."""
+        aux_hidden_format = getattr(
+            format_provider,
+            "dspark_aux_hidden_state_format",
+            None,
+        )
```
[the remark this answers is not in the document]
Comment (qijiajin) [reply vllm_ascend/worker/v2/spec_decode/dspark/speculator.py:155]: `getattr(format_provider, name, default)` is safe when `format_provider is None`; Python returns the supplied default rather than raising. The following `architectures` and `model_type` lookups also use three-argument `getattr`, so this path resolves to `None` and returns without touching the target. I rechecked this path while running the focused suite (48 passed).
